iOS-TvShows/iOS-TvShows/Modules/Shows/Components/TableViewCell/ShowTableViewCell.swift

```diff
@@ -0,0 +1,77 @@
+//
+//  ShowTableViewCell.swift
+//  iOS-TvShows
+//
+//  Created by Filip Ban on 24.11.2021..
+//
+
+import UIKit
+import Kingfisher
+import RxCocoa
+
+struct ShowTableCellItem: TableCellItem {
+    let show: Show
+    let didSelect: ((ShowTableCellItem) -> Void)?
+}
+
+extension ShowTableCellItem {
+    func cell(from tableView: UITableView, at indexPath: IndexPath) -> UITableViewCell {
+        let cell = tableView.dequeueReusableCell(ofType: ShowTableViewCell.self, for: indexPath)
+        cell.setData(with: show)
+        return cell
+    }
+
+    var height: CGFloat {
+        114
+    }
+
+    func didSelect(at indexPath: IndexPath) {
+        didSelect?(self)
+    }
+}
+
+class ShowTableViewCell: UITableViewCell {
+
+    var customImageView: UIImageView!
+    var label: UILabel!
+
+    override init(style: UITableViewCell.CellStyle, reuseIdentifier: String?) {
+        super.init(style: style, reuseIdentifier: reuseIdentifier)
+
+        backgroundColor = UIColor.TVShows.appWhite
+        customImageView = UIImageView()
+        contentView.addSubview(customImageView)
+        label = UILabel(with: UIFont.systemFont(ofSize: 17))
+        contentView.addSubview(label)
+
+        customImageView.clipsToBounds = true
+        customImageView.roundAllCorners(withRadius: 4)
+
+        customImageView.snp.makeConstraints {
+            $0.size.equalTo(CGSize(width: 64, height: 90))
+            $0.centerY.equalToSuperview()
+            $0.leading.equalToSuperview().inset(16)
+        }
+
+        label.snp.makeConstraints {
+            $0.leading.equalTo(customImageView.snp.trailing).offset(24)
+            $0.trailing.equalToSuperview().inset(24)
+            $0.centerY.equalToSuperview()
+        }
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    func setData(with show: Show) {
+        let placeholderImage = UIImage.TVShows.Icons.showPlaceholder
+        label.text = show.title
+
+        guard let url = URL(string: show.imageUrl ?? "") else {
+            customImageView.image = placeholderImage
+            return
+        }
+        customImageView.kf.setImage(with: url, placeholder: placeholderImage, options: nil, completionHandler: nil)
```

Comment:
This is a bit nitpicking now, but consider this:

```swift
    func setData(with show: Show) {
        let placeholderImage = UIImage.TVShows.Icons.showPlaceholder
        label.text = show.title
        customImageView.kf.setImage(
            with: show.imageUrl.flatMap(URL.init),
            placeholder: placeholderImage
        )
    }
```

This is a bit improved code, due to two facts:

1. we don't have the guard anymore:
- guard in the middle of the func is problematic because you are introducing some kind of a GOTO statement which could represent problem in debugging. We prefer having returns only on top of the function or the bottom, so that the code is more streamlined and easier to follow
- what if you forgot adding `customImageView.image = placeholderImage`, or someone deletes it - it will lead to bug where in case if the next show doesn't have the image, the reused cell will display the image of the previous one. Kingfisher does that automatically for you when calling setImage(with: nil, placeholderImage: myPlaceholder) - it will use placeholder image in that case
2. It's shorter due to higher order function on `Optional<String>` which will convert String to URL if one exists ;)